chronos/asyncfutures2.nim

```diff
@@ -611,7 +609,8 @@ proc `or`*[T, Y](fut1: Future[T], fut2: Future[Y]): Future[void] =
   ##
   ## If cancelled, ``fut1`` and ``fut2`` futures WILL NOT BE cancelled.
   var retFuture = newFuture[void]("chronos.or")
-  proc cb(udata: pointer) {.gcsafe.} =
+  var cb: proc(udata: pointer) {.gcsafe, raises: [Defect].}
```

Comment:
Why this continuation callback needs explicit `raises: [Defect]`. Why not mark every possible continuation callback as `raises:[Defect]`?

Reply:
nim bug: https://github.com/nim-lang/Nim/issues/17369